src/common/include/common/keymap.h

```diff
@@ -3,10 +3,33 @@
 #ifndef _COMMON_KEYMAP_H
 #define _COMMON_KEYMAP_H
 
+#include <stdbool.h>
 #include <stdint.h>
 
+// Keymap defined by board
 #if defined(KM_LAY) && defined(KM_OUT) && defined(KM_IN)
-extern uint16_t __xdata KEYMAP[KM_LAY][KM_OUT][KM_IN];
+    extern uint16_t __code KEYMAP[KM_LAY][KM_OUT][KM_IN];
+    extern uint16_t __xdata DYNAMIC_KEYMAP[KM_LAY][KM_OUT][KM_IN];
+    #define HAVE_KEYMAP 1
+#else
+    #define HAVE_KEYMAP 0
```

Comment:
Should this be an error? Compilation will fail if `KM_*` are not defined.

Reply:
Good idea. So long as the AVR boards still build. They shouldn't be including keymap.h anyways

Reply:
Erm, it can't be done this way. common/keymap.c includes some functions that are always present even on AVR boards